### PR TITLE
K8SPSMDB-190 create the backup cr with curl

### PR DESCRIPTION
**Pull request by phin1x**
create the backup cr with curl instead of kubectl

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/percona-server-mongodb-operator?pullRequest=353) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-server-mongodb-operator?pullRequest=353) before we can accept your contribution.<br/><hr/>**Fabian Weber** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-server-mongodb-operator?pullRequest=353) it.</sub>

File: pkg/psmdb/backup/job.go [hunk 1]
```diff
@@ -86,29 +88,6 @@ func NewBackupCronJobLabels(crName string) map[string]string {
...
-				  labels:
-				    ancestor: "` + backup.Name + `"
-				    cluster: "${psmdbCluster}"
-				    type: "cron"
-				spec:
-				  psmdbCluster: "${psmdbCluster}"
-				  storageName: "` + backup.StorageName + `"
-			EOF
-		`,
+		`curl -vvv -X POST --cacert /run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Content-Type: application/json" -H "Authorization: Bearer $(cat /run/secrets/kubernetes.io/serviceaccount/token)" --data "{\"kind\":\"PerconaServerMongoDBBackup\",\"apiVersion\":\"psmdb.percona.com/v1\",\"metadata\":{\"generateName\":\"cron-${psmdbCluster:0:16}-$(date -u "+%Y%m%d%H%M%S")-\",\"labels\":{\"` + backup.Name + `\":\"testing\",\"cluster\":\"${psmdbCluster}\",\"type\":\"cron\"}},\"spec\":{\"psmdbCluster\":\"${psmdbCluster}\",\"storageName\":\"` + backup.StorageName + `\"}}" https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}/apis/psmdb.percona.com/v1/namespaces/${NAMESPACE}/perconaservermongodbbackups`,
```
Comment (delgod) [inline pkg/psmdb/backup/job.go:91]: 1-liner is not really readable :(
can we improve it?
```suggestion
		`curl \
		    -vvv \
		    -X POST \
		    --cacert /run/secrets/kubernetes.io/serviceaccount/ca.crt \
		    -H "Content-Type: application/json" \
		    -H "Authorization: Bearer $(cat /run/secrets/kubernetes.io/serviceaccount/token)" \
		    --data "{
		        \"kind\":\"PerconaServerMongoDBBackup\",
		        \"apiVersion\":\"psmdb.percona.com/v1\",
		        \"metadata\":{
		            \"generateName\":\"cron-${psmdbCluster:0:16}-$(date -u "+%Y%m%d%H%M%S")-\",
		            \"labels\":{
		                \"ancestor\": \"` + backup.Name + `\",
		                \"cluster\":\"${psmdbCluster}\",
		                \"type\":\"cron\"
		            }
		        },
		        \"spec\":{
		            \"psmdbCluster\":\"${psmdbCluster}\",
		            \"storageName\":\"` + backup.StorageName + `\"
		        }
		    }" \
		    https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}/apis/psmdb.percona.com/v1/namespaces/${NAMESPACE}/perconaservermongodbbackups`,
```

Comment (phin1x) [reply pkg/psmdb/backup/job.go]: Sure, curl is formatted

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| security-context | failed |
| arbiter | passed |
| scheduled-backup | failed |
| operator-self-healing | passed |
| service-per-pod | passed |
| liveness | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | failed |
| upgrade | passed |
| one-pod | passed |
| self-healing | passed |
| limits | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| scaling | failed |
| demand-backup | failed |
| security-context | failed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| scheduled-backup | failed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| scaling | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| security-context | failed |
| operator-self-healing | passed |
| arbiter | passed |
| scheduled-backup | failed |
| service-per-pod | failed |
| liveness | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| upgrade | failed |
| storage | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| self-healing | failed |
| scaling | failed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| security-context | failed |
| operator-self-healing | passed |
| arbiter | passed |
| scheduled-backup | failed |
| service-per-pod | passed |
| liveness | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| init-deploy | passed |
| limits | passed |
| scaling | failed |
| one-pod | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| upgrade | passed |
| security-context | failed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| operator-self-healing | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| storage | passed |
| service-per-pod | passed |
| self-healing | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| one-pod | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | failed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| limits | passed |
| monitoring | passed |
| self-healing | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| security-context | failed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | failed |

**Comment by tplavcic** [comment]
Thank you for your contribution!

**Pull request merged**